src/resource/user/user.service.ts

```diff
@@ -17,4 +21,23 @@ export class UserService {
   async findByUsername(_username: string) {
     return this.prisma.user.findFirst({ where: { username: _username } });
   }
+
+  async createPasswordResetLink(_id: string) {
+    const { createHmac, randomBytes } = await import('crypto');
+    const secret = process.env.CRYPTO_SECRET;
+    const token = randomBytes(32).toString('hex');
+    const resetToken = createHmac('sha256', secret).update(token).digest('hex');
```

Comment:
Since we are not really reusing the secret here, probably can shorten this to use `process.env.CTOPTO_SECRET` directly. Unless there's a different reason to keep it. 👌 

Or it can be there to make the code readable? which makes sense too.

Reply:
This has been updated